### ISSUE TITLE
[Bug]: fn_encounter_postAmbush.sqf Undefined variable _originposition creates infinite loop

### ISSUE THREAD

**Issue by magssibbert** [igorkis-scrts/A3-Antistasi-Plus#80]
### Describe the bug

23:21:08 File /documents/A3-Antistasi-Plus/A3-Antistasi/SCRT/Encounter/fn encounter_postAnbush.sqf..., line 17

23:21:00 Error in expression <_road findIf (position x) distance2D _originPosition> 500) != -1)) exitwith > 
23:21:00 Error position: <_originPosition> 500) != -1)) exitwith >
23:21:00 Error Undefined variable in expression: originposition 

The above is looping infinitly, massivly lagging the server. Server restart fixed it.

I would expect there to be a failsafe in the while{true} loop

![unknown](https://user-images.githubusercontent.com/45903063/189774284-845207a5-8412-4837-b330-76cff841fa2a.png)
23:21:00 File /documents/A3-Antistasi-Plus/A3-Antistasi/SCRT/Encounter/fn_encounter postAmbush.sqf..., line 17

### How to reproduce

Have _originposition be undefined.

Unfortunaly since logging wasnt enabled on the (Linux) Server, I cant say what caused that. It went unnoticed for atleast 30mins. 

### Version

1.8.3.2 - 08.09.22

### Have you altered the code?

No

### Map

Cam Lao Nam

### What server?

Private dedicated server

### Mods

```html
-mod=vn -mod=mods/@cba_a3 -mod=mods/@ace3 -mod=mods/@ace_sog -mod=mods/@acre2 -mod=mods/@acre2_sog -mod=mods/@aaren_tracer_fix -mod=mods/@adv_breaching -mod=mods/@adv_rappelling_re -mod=mods/@adv_towing -mod=mods/@adv_zipline -mod=mods/@a3forces_med -mod=mods/@bp_on_chest -mod=mods/@blastcore_server -mod=mods/@grad_trenches -mod=mods/@immersion_cigs -mod=mods/@splendid_smoke -mod=mods/@survivable_crashes -mod=mods/@suppress -mod=mods/@swim_faster -mod=mods/@zeus_enhanced -mod=mods/@zeus_enhanced_a3 -mod=mods/@camofaces -mod=mods/@unsung_redux_stable -mod=mods/@sog_melee_key -mod=mods/@ares
```

**Comment by igorkis-scrts** [comment]
Thanks for report. I guess there were no players on server at some point and event loop suspending didn't work as expectedly.
I already have early escapes in case when there is empty position in the new types of events, I guess other events should have it too.
Your case will be fixed in the upcoming update, thanks for report!

**Comment by igorkis-scrts** [comment]
Fixed in 2.0.
